src/App.js

```diff
@@ -8,29 +8,35 @@ import { mentors } from "./mentors.json";
 import "./App.css";
 
 const visibleMentorIds = mentors.map((mentor, index) => index);
+
+// Convert names into usable `#`-ids.
+const mentorNames = mentors.map(({ name }) =>
+  name.toLowerCase().replace(" ", "-").replace("(", "").replace(")", "")
+);
 const setHash = (hash) => window.history.replaceState({}, "", `#${hash}`);
 
 function App() {
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [activeMentorId, setActiveMentorId] = useState(0);
 
   useEffect(() => {
-    // Checks hash and ensures that any modal with a corresponding ID is open.
+    // Checks hash and ensures that any modal with a corresponding name is open.
     let ensureModalFromHash = () => {
       let hash = window.location.hash.slice(1);
-      let num = hash.length ? Number(hash) : "";
+      let mentorName = hash.length ? String(hash) : "";
+      let mentorId = mentorName.length ? mentorNames.indexOf(mentorName) : -1;
 
-      if (typeof num == "number" && !isNaN(num) && mentors[num]) {
-        if (!isModalOpen || num !== activeMentorId) {
-          activateModal(num);
+      if (mentorId !== -1) {
+        if (!isModalOpen || mentorId !== activeMentorId) {
```

Comment:
Feels like this check should be done in `activateModal` instead, what do you think?

Reply:
I guess it could, although these cases would only ever occur here. I have no strong opinions: feel free to add it if you want.

Reply:
Hmm alright, I guess it's not a big issue.